Add argument to convertUI that forces full UI rebuild

krzywon opened the pull request:
## Description

This adds new arguments to convertUI.py and setup.py to force all UI elements to be rebuilt, regardless if they exist or not. This is partly in response to the cleanup needed to test #2478 but could also have utility for future dependency changes.

## How Has This Been Tested?

Ran `python convertUI.py -f`, `python setup.py rebuild_ui`, and `python setup.py clean`, and all 3 rebuilt the UI elements regardless if they existed or not.

## Review Checklist (please remove items if they don't apply):

- [ ] Code has been reviewed
- [ ] Functionality has been tested
- [ ] Windows installer (GH artifact) has been tested (installed and worked) 
- [ ] MacOSX installer (GH artifact) has been tested (installed and worked) 
- [ ] User documentation is available and complete (if required)
- [ ] Developers documentation is available and complete (if required)
- [ ] The introduced changes comply with SasView license (BSD 3-Clause)

lucas-wilkins commented:
I think it should be part of "clean".

rozyczko commented:
> I think it should be part of "clean".

Should it?
I thought `clean` should delete all setup generated files. The proposed flag is to enforce the `.ui` -> `.py` rebuild, thus generating dependent python files.

rozyczko commented on src/sas/qtgui/convertUI.py:
```diff
@@ -48,13 +51,17 @@ def file_in_newer(file_in, file_out):
             file_in = os.path.join(root, file)
             file_out = os.path.splitext(file_in)[0]+'.py'
-            if file_in_newer(file_in, file_out):
+            if force_recreate or file_in_newer(file_in, file_out):
```
this conditional is evaluated three times. It may be easier to pre-evaluate it.
```
redo = force_recreate or file_in_newer(file_in, file_out)
...
if redo:
...
```

krzywon replied:
`file_in_newer` passes different files for `file_in` and `file_out` every time it is run, so this would create an invalid check. I could probably incorporate the later checks into this loop. Let me look into that.

rozyczko reviewed (approved):
The code looks good, minor stylistic points raised.

lucas-wilkins commented:
> Should it?
> I thought `clean` should delete all setup generated files. The proposed flag is to enforce the `.ui` -> `.py` rebuild, thus generating dependent python files.

I could have been clearer.

Removing existing automatic python files should be "clean", and that should be enough to trigger a rebuild when you run sasview.

However, it might also be good to have the actual making of the py files in build/rebuild

krzywon commented:
I tried to clean up `convertUI.py` but the Ubuntu build failed so I reverted the changes and should probably not put any more time into it. I also incorporated the change suggested by @lucas-wilkins and the arguments `clean` and `rebuild_ui` now force the GUI to rebuild.

lucas-wilkins commented:
I've noticed that in main, automatic UI rebuilds don't seem to be happening.

lucas-wilkins commented:
I fixed that, hope the changes don't get in the way
